stmhal/uart.c

```diff
@@ -115,7 +115,7 @@ void uart_deinit(void) {
 STATIC bool uart_init2(pyb_uart_obj_t *uart_obj) {
     USART_TypeDef *UARTx;
     IRQn_Type irqn;
-    uint32_t GPIO_Pin, GPIO_Pin2;
+    uint32_t GPIO_Pin, GPIO_Pin2= NULL;
```

Comment:
make DEBUG=1 causes the same error (even for the pyboard).

I'm going to do a PR just for that change, and add DEBUG=1 builds to my set of travis builds.